src/Interpreters/InterpreterSystemQuery.cpp

```diff
@@ -1134,10 +1135,13 @@ void InterpreterSystemQuery::dropDatabaseReplica(ASTSystemQuery & query)
         throw Exception(ErrorCodes::LOGICAL_ERROR, "Invalid query");
 }
 
-bool InterpreterSystemQuery::trySyncReplica(IStorage * table, SyncReplicaMode sync_replica_mode, const std::unordered_set<String> & src_replicas, ContextPtr context_)
+bool InterpreterSystemQuery::trySyncReplica(StoragePtr table, SyncReplicaMode sync_replica_mode, const std::unordered_set<String> & src_replicas, ContextPtr context_)
  {
     auto table_id_ = table->getStorageID();
-    if (auto * storage_replicated = dynamic_cast<StorageReplicatedMergeTree *>(table))
+    if (auto * storage_mv = dynamic_cast<StorageMaterializedView *>(table.get()))
```

Comment:
seems to be while here is better? to handle dependant mv.

Reply:
Added a loop.

It wouldn't be dependent MVs, but an MV that writes `TO` another mv:
```sql
CREATE MATERIALIZED VIEW mv2 TO mv1 AS SELECT ...;
```
It's a little weird, but I guess may be used in practice to send data from multiple source tables into the same destination table. One MV would own the table, other MVs would say `TO mv1`. (It's more symmetric to create the destination table separately and make all MVs say `TO dest` instead, but maybe someone would prefer the other style.)